File: app/lib/decision/buildInsight.ts
// app/lib/decision/buildInsight.ts

import { deriveTechnicalEvidence } from "./deriveTechnicalEvidence";
import { technicalLens } from "./technicalLens";
import { fundamentalLens } from "./fundamentalLens";
import { marketLens } from "./marketLens";
import { aggregateDecision } from "./decisionEngine";
import { getBusinessArchetype } from "./businessArchetype";
import { ContextualEvidence } from "./contextualEvidence";

export async function buildInsight(
  symbol: string,
  data: {
    candles: number[];        // stock close prices
    indexCandles: number[];   // benchmark close prices (NIFTY, etc.)
    fundamental: any;
    market: any;
    contextualEvidence: ContextualEvidence[];
  }
) {
  // Business identity (deterministic)
  const archetype = getBusinessArchetype(symbol);

  // ✅ FIXED: pass BOTH arguments
  const technicalEvidence = deriveTechnicalEvidence(
    data.candles,
    data.indexCandles
  );

  const technical = technicalLens(technicalEvidence);
  const fundamental = fundamentalLens(data.fundamental);
  const market = marketLens(data.market);

  const final = aggregateDecision(
    technical,
    fundamental,
    market,
    archetype
  );

  return {
    symbol,
    archetype,

    technical,
    fundamental,
    market,

    contextualEvidence: data.contextualEvidence,

    aiCommentary:
      `Decision based on canonical technical facts and ${archetype} structure.`,

    finalAction: final.action,
    finalConfidence: final.confidence,
    finalRationale: final.rationale,
  };
}

File: app/lib/decision/buildInsight.cs
import { aggregateDecision } from "./decisionEngine";
import { technicalLens } from "./technicalLens";
import { fundamentalLens } from "./fundamentalLens";
import { marketLens } from "./marketLens";
import {import { generateAICommentary } from "./aiCommentary";
    symbol,
    technical,
    fundamental,
    market,
    aiCommentary: generateAICommentary({
      symbol,
      sector: data.sector,
      context: data.context,
    }),
    finalAction: action,
    finalRationale: rationale,
  };
}

export function buildInsight(symbol: string, data: any) {
  const technical = technicalLens(data.technical);
  const fundamental = fundamentalLens(data.fundamental);
  const market = marketLens(data.market);

  const { action, rationale } = aggregateDecision(
    technical,
    fundamental,
    market
  );

